openedx/core/lib/api/paginators.py

```diff
@@ -29,6 +30,38 @@ def get_paginated_response(self, data):
         })
 
 
+class NamespacedPageNumberPagination(pagination.PageNumberPagination):
+    """
+    Pagination scheme that returns results with pagination metadata
+    embedded in a "pagination" attribute.  Can be used with data
+    that comes as a list of items, or as a dict with a "results"
+    attribute that contains the code.
+    """
+
+    page_size_query_param = "page_size"
+
+    def get_paginated_response(self, data):
+        """
+        Annotate the response with pagination information
+        """
+        metadata = OrderedDict([
+            ('next', self.get_next_link()),
+            ('previous', self.get_previous_link()),
+            ('count', self.page.paginator.count),
+            ('num_pages', self.page.paginator.num_pages),
+        ])
+        if isinstance(data, dict):
+            if 'results' not in data:
+                raise TypeError(u'Malformed result dict')
+            data['pagination'] = metadata
+        else:
+            data = OrderedDict([
```

Comment:
DRF generally returns `OrderedDict`s. I didn't see a reason to break from their practice.  It provides a slightly more predictable response through the API, and putting the `results` first foregrounds the primary information the API caller is looking for.  Foregrounding that information is in line with the [API guidelines](https://openedx.atlassian.net/wiki/display/AC/edX+REST+API+Conventions#edXRESTAPIConventions-HighLevelRequirements).  See the "Simple" bulletpoint undller "High Level Requirements."

That said, it's not a hard requirement, but I can't think of a compelling reason not to order it.